checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,375 @@
+  public TransferResult<CFValue, CFStore> visitMethodInvocation(
+      MethodInvocationNode n, TransferInput<CFValue, CFStore> in) {
+    TransferResult<CFValue, CFStore> result = super.visitMethodInvocation(n, in);
+
+    updateStoreWithTempVar(result, n);
+    if (!atypeFactory.getChecker().hasOption(MustCallChecker.NO_ACCUMULATION_FRAMES)) {
+      List<JavaExpression> targetExprs =
+          getCreatesObligationExpressions(n, atypeFactory, atypeFactory);
+      for (JavaExpression targetExpr : targetExprs) {
+        AnnotationMirror defaultType =
+            atypeFactory
+                .getAnnotatedType(TypesUtils.getTypeElement(targetExpr.getType()))
+                .getAnnotationInHierarchy(atypeFactory.TOP);
+
+        if (result.containsTwoStores()) {
+          CFStore thenStore = result.getThenStore();
+          lubWithStoreValue(thenStore, targetExpr, defaultType);
+
+          CFStore elseStore = result.getElseStore();
+          lubWithStoreValue(elseStore, targetExpr, defaultType);
+        } else {
+          CFStore store = result.getRegularStore();
+          lubWithStoreValue(store, targetExpr, defaultType);
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Computes the LUB of the current value in the store for expr, if it exists, and defaultType.
+   * Inserts that LUB into the store as the new value for expr.
+   *
+   * @param store a CFStore
+   * @param expr an expression that might be in the store
+   * @param defaultType the default type of the expression's static type
+   */
+  private void lubWithStoreValue(CFStore store, JavaExpression expr, AnnotationMirror defaultType) {
+    CFValue value = store.getValue(expr);
+    CFValue defaultTypeAsCFValue =
+        analysis.createSingleAnnotationValue(defaultType, expr.getType());
+    CFValue newValue;
+    if (value == null) {
+      newValue = defaultTypeAsCFValue;
+    } else {
+      newValue = value.leastUpperBound(defaultTypeAsCFValue);
+    }
+    store.clearValue(expr);
+    store.insertValue(expr, newValue);
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitObjectCreation(
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitTernaryExpression(
+      TernaryExpressionNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitTernaryExpression(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  /**
+   * Adds newAnno as the value for target to all stores contained in result.
+   *
+   * @param result a TransferResult containing one or more stores
+   * @param target a JavaExpression whose type is being modified
+   * @param newAnno the new type for target
+   */
+  public void insertIntoStores(
```

Comment:
This same method appears in the accumulation transfer function.  How about moving in into their shared superclass?